packages/vm/core/errors/impl.go

```diff
@@ -0,0 +1,74 @@
+package errors
+
+import (
+	"github.com/iotaledger/wasp/packages/iscp"
+	"github.com/iotaledger/wasp/packages/kv/codec"
+	"github.com/iotaledger/wasp/packages/kv/dict"
+	"github.com/iotaledger/wasp/packages/kv/kvdecoder"
+	"github.com/iotaledger/wasp/packages/vm/core/errors/commonerrors"
+	"github.com/iotaledger/wasp/packages/vm/vmerrors"
+	"math"
+)
+
+var Processor = Contract.Processor(initialize,
+	FuncRegisterError.WithHandler(funcRegisterError),
+	FuncGetErrorMessageFormat.WithHandler(funcGetErrorMessageFormat),
+)
+
+func initialize(ctx iscp.Sandbox) dict.Dict {
+	ctx.Log().Debugf("errors.initialize.success hname = %s", Contract.Hname().String())
+	return nil
+}
+
+func funcRegisterError(ctx iscp.Sandbox) dict.Dict {
+	ctx.Log().Debugf("Registering error")
+	e := NewStateErrorCollectionWriter(ctx.State(), ctx.Caller().Hname())
+
+	params := kvdecoder.New(ctx.Params())
+	errorMessageFormat := params.MustGetString(ParamErrorMessageFormat)
+
+	if len(errorMessageFormat) == 0 {
+		panic(commonerrors.ErrMessageFormatEmpty)
+	}
```

Comment:
```suggestion
	ctx.Require(len(errorMessageFormat) > 0, commonerrors.ErrMessageFormatEmpty))
```

although I'm not sure the types will match :stuck_out_tongue: